src/main/java/com/ably/kafka/connect/utils/RecordHeaderConversions.java

```diff
@@ -0,0 +1,136 @@
+package com.ably.kafka.connect.utils;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.internal.LinkedTreeMap;
+import io.ably.lib.types.MessageExtras;
+import io.ably.lib.util.JsonUtils;
+import org.apache.kafka.connect.header.Header;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.util.Base64;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class RecordHeaderConversions {
+
+    private static final Logger logger = LoggerFactory.getLogger(RecordHeaderConversions.class);
+    private static final String PUSH_HEADER = "com.ably.extras.push";
+    private static final String KAFKA_KEY = "kafka";
+    private static final String PUSH_KEY = "push";
+    /**
+     * Returns the extras object to use when converting a Kafka message
+     * to an Ably message.
+     * <p>
+     * If the Kafka message has a key, it is base64 encoded and set as the
+     * "key" field in the extras and is added under "kafka" extras.
+     * <p>
+     * If the Kafka message has headers, they are set as the "headers" field
+     * in the extras.
+     *
+     * If the Kafka message has "com.ably.extras.push" header this is set as "push" extras for Ably message.
+     *
+     * @param record The sink record representing the Kafka message
+     * @return The Kafka message extras object
+     */
+    @Nullable
+    public static MessageExtras toMessageExtras(final SinkRecord record) {
+        JsonUtils.JsonUtilsObject kafkaExtras = null;
+        JsonUtils.JsonUtilsObject extrasObject = null;
+
+        Object pushExtras = null;
+        final byte[] key = (byte[]) record.key();
+        if (key != null) {
+            kafkaExtras = JsonUtils.object();
+            final String keyValue = Base64.getEncoder().encodeToString(key);
+            kafkaExtras.add("key", keyValue);
+        }
+
+        if (record.headers().isEmpty()){
```

Comment:
Missing a space before `{` here

Reply:
Hopefully that's also addressed as part of 1918c37cf4bf6bb6c23d26869153db9b0ba24eba